Extractor/Pushers/CDFPusher.cs

```diff
@@ -1253,18 +1253,18 @@ private async Task MarkReferencesAsDeleted(IEnumerable<string> externalIds, Canc
         /// </summary>
         private async Task EnsureConfigInit(CancellationToken token)
         {
-            if (!config.DataSetId.HasValue && !string.IsNullOrEmpty(config.DataSetExternalId))
+            if (config.DataSet != null)
             {
                 try
                 {
-                    var dataSet = await destination.CogniteClient.DataSets.RetrieveAsync(new[] { config.DataSetExternalId }, false, token);
-                    config.DataSetId = dataSet.First().Id;
+                    var id = await destination.CogniteClient.DataSets.GetId(config.DataSet, token);
+                    config.DataSet.Id = id;
                 }
                 catch (ResponseException ex)
                 {
                     log.LogError("Could not fetch data set by external id. It may not exist, or the user may lack" +
                         " sufficient access rights. Project {Project} at {Host}, id {Id}: {Message}",
-                        config.Project, config.Host, config.DataSetExternalId, ex.Message);
+                        config.Project, config.Host, config.DataSet.ExternalId, ex.Message);
```

Comment:
```suggestion
                        config.Project, config.Host, config.DataSet?.ExternalId, ex.Message);
```
This one might be unnecessary if this section is reworked.